dataPipelines/gc_thumbnails/utils.py

```diff
@@ -1,26 +1,48 @@
 from pathlib import Path
 import fitz
 import typing as t
+import multiprocessing
+from concurrent.futures import ProcessPoolExecutor
+
 
 class ThumbnailsCreator:
     def __init__(
         self,
         input_directory: t.Union[str, Path],
-        output_directory: t.Union[str, Path]
+        output_directory: t.Union[str, Path],
+        max_threads: int
```

Comment:
Same note about arg name, probably better to call it `max_workers` since it's not actually threads